### PR TITLE
DM-28600: Add option to apply skyCorr to ProcessBrightStarsTask

### PR DESCRIPTION
**Pull request by MorganSchmitz**
[Jira ticket](https://jira.lsstcorp.org/browse/DM-28600)

Jenkins run [#33511](https://ci.lsst.codes/blue/organizations/jenkins/stack-os-matrix/detail/stack-os-matrix/33511/pipeline)

### CODE REVIEW

File: python/lsst/pipe/tasks/processBrightStars.py [hunk 1]
```diff
@@ -348,6 +382,8 @@ def run(self, inputExposure, refObjLoader=None, dataId=None):
         dataId : `dict` or `lsst.daf.butler.DataCoordinate`
             The dataId of the exposure (and detector) bright stars should be
             extracted from.
+        skyCorr : `lsst.afw.math.backgroundList.BackgroundList`
```
Comment (laurenam) [inline python/lsst/pipe/tasks/processBrightStars.py:385]: ...or `None`, optional

https://developer.lsst.io/python/numpydoc.html#optional-parameters

File: python/lsst/pipe/tasks/gen2measureExtendedPsf.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
+# This file is part of pipe_tasks.
```
Comment (laurenam) [inline python/lsst/pipe/tasks/gen2measureExtendedPsf.py:1]: I'm confused about the addition of this file in the context of this ticket?

File: python/lsst/pipe/tasks/processBrightStars.py [hunk 3]
```diff
@@ -134,6 +145,11 @@ class ProcessBrightStarsConfig(pipeBase.PipelineTaskConfig,
             " annular flux.",
         default=('BAD', 'CR', 'CROSSTALK', 'EDGE', 'NO_DATA', 'SAT', 'SUSPECT', 'UNMASKEDNAN')
     )
+    doApplySkyCorr = pexConfig.Field(
+        dtype=bool,
+        doc="Apply sky correction before extracting stars?",
```
Comment (laurenam) [inline python/lsst/pipe/tasks/processBrightStars.py:150]: Can you be a bit more specific about which "sky correction" you are referring to here (even just adding "full focal plane" would help...there...are...just...so...many...!!!)?

File: python/lsst/pipe/tasks/gen2measureExtendedPsf.py (hunk 2 above)
Comment (MorganSchmitz) [reply python/lsst/pipe/tasks/gen2measureExtendedPsf.py]: Oh yeah, that's just me failing hard at git. Removed it. Sorry!

File: python/lsst/pipe/tasks/processBrightStars.py (hunk 3 above)
Comment (laurenam) [reply python/lsst/pipe/tasks/processBrightStars.py]: Your update snuck into the bug fix commit 😉

File: python/lsst/pipe/tasks/processBrightStars.py [hunk 4]
```diff
@@ -184,6 +200,24 @@ def __init__(self, butler=None, initInputs=None, *args, **kwargs):
...
             self.makeSubtask('refObjLoader', butler=butler)
 
+    def applySkyCorr(self, calexp, skyCorr):
+        """Apply correction to the sky background level
+        Sky corrections can be generated with the 'skyCorrection.py'
+        executable in pipe_drivers. Because the sky model used by that
+        code extends over the entire focal plane, this can produce
+        better sky subtraction.
+        The calexp is updated in-place.
+        Parameters
```
[the remark this answers is not in the document]
Comment (laurenam) [reply python/lsst/pipe/tasks/processBrightStars.py:223]: Still need a space after title description.

File: python/lsst/pipe/tasks/processBrightStars.py [hunk 5]
```diff
@@ -183,8 +199,8 @@ class ProcessBrightStarsTask(pipeBase.PipelineTask, pipeBase.CmdLineTask):
     def __init__(self, butler=None, initInputs=None, *args, **kwargs):
         super().__init__(*args, **kwargs)
         # Compute (model) stamp size depending on provided "buffer" value
-        self.modelStampSize = (int(self.config.stampSize[0]*self.config.modelStampBuffer),
-                               int(self.config.stampSize[1]*self.config.modelStampBuffer))
+        self.modelStampSize = [int(self.config.stampSize[0]*self.config.modelStampBuffer),
+                               int(self.config.stampSize[1]*self.config.modelStampBuffer)]
```
Comment (laurenam) [inline python/lsst/pipe/tasks/processBrightStars.py:203]: I’m fine with this bug fix here and would keep it as it’s own commit, but could you split out the unrelated doc update and add a note in the commit message describing the bug that needed fixing?

**Review by laurenam — APPROVED** [review]

**Pull request merged**